### PR TITLE
Made improvements to combinations.py

### PR DESCRIPTION
**Pull request by Rolv-Apneseth**
I have made some improvements to combinations.py.

Added:
* A description at the top, which includes a wikipedia link
* Type hints to the combinations function
* Some more doctests, which include 0 and negative numbers
*  A check within the function so the function does not attempt to get factorials of negative numbers
* Some print statements in the main block which show some practical applications of the function

If you would like anything else, or if you don't like the changes, please let me know. Thanks!

### CODE REVIEW

**Comment by Rolv-Apneseth** [comment]
In relation to #3682

File: maths/combinations.py [hunk 1]
```diff
@@ -1,19 +1,66 @@
+"""
+This function will return the total number of different combinations of
+k length which can be made from n values.
+
+Wikipedia page here: https://en.wikipedia.org/wiki/Combination
+
+For doctests, run: `python3 -m doctest -v combinations.py`
+
+Run `python3 combinations.py` for print statements which demonstrate use
+cases for this function.
```
Comment (shellhub) [inline maths/combinations.py:10]:
```suggestion
https://en.wikipedia.org/wiki/Combination
```

Comment (Rolv-Apneseth) [reply maths/combinations.py:10]: So it's better to take out the description at the start?

**Review by shellhub — COMMENTED** [review]
Correct!

File: maths/combinations.py [hunk 2]
```diff
@@ -1,19 +1,66 @@
...
+
+    >>> combinations(-4, -5)
+    Invalid value for n and/or k given.
+    -1
     """
+
+    # If either of the conditions are true, the function is being asked
+    # to calculate a factorial of a negative number, which is not possible
+    if n < k or k < 0:
+        print("Invalid value for n and/or k given.")
```
[the remark this answers is not in the document]
Comment (Rolv-Apneseth) [reply maths/combinations.py:36]: I'm kind of new to doctests in general, how do I make the doctest for an exception?

[the remark this answers is not in the document]
Comment (Rolv-Apneseth) [reply maths/combinations.py]: Nvm rhink I got it

File: maths/combinations.py [hunk 3]
```diff
@@ -1,19 +1,57 @@
...
     15504
+
+    >>> combinations(52, 5)
+    2598960
+
+    >>> combinations(0, 0)
+    1
+
+    >>> combinations(-4, -5)
+    Traceback (most recent call last):
```
Comment (shellhub) [inline maths/combinations.py:29]:
```suggestion
    ...
    Traceback (most recent call last):
```

**Review by shellhub — APPROVED** [review]

**Comment by shellhub** [comment]
@Rolv-Apneseth Thanks for your contribution :)

**Comment by Rolv-Apneseth** [comment]
No problem!

**Pull request merged**